src/differentiation.jl

```diff
@@ -23,12 +23,17 @@ struct FiniteDifference{T <: AbstractFloat} <: DifferentiationMethod
     ε::T
 end
 
-# Default constructors
-FiniteDifference{T}() where {T <: AbstractFloat} = FiniteDifference{T}(T(1e-6))
-FiniteDifference() = FiniteDifference{Float64}()
+# If ε not specified, default to 1e-6
+FiniteDifference() = FiniteDifference(1e-6)
```

Comment:
I think this `FiniteDifference` constructor is a carry-over from before the recent commits; see the updated replacement on the left side of the diff.